### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by jackzampolin**
This pr adds a `Dockerfile` that builds the project. It also adds a `make docker` target to build the image. Additional PRs with commands to run the docker image will be pushed at a later time.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+WORKDIR /go/src/github.com/cosmos/ethermint
+
+# Install dependencies
+RUN apk add --update $PACKAGES
+
+# Add source files
+COPY . .
+
+# Make the binary
+RUN make update-tools get-vendor-deps build
```
Comment (zramsay) [inline Dockerfile:16]: oi, let's use `_` instead of `-` for harmony with rest of stack

Comment (alexanderbez) [reply Dockerfile:16]: Must we? The SDK is a mixed bag of both and GNU convention is to use `-` afaik.

Comment (zramsay) [reply Dockerfile:16]: i'd say at least for `make get_vendor_deps` and any targets that are the same in multiple Makefiles across the stack. The targets that use `-` are all the networks stuff from greg (somewhat seperate than the core make targets)

**Review by alexanderbez — CHANGES_REQUESTED** [review]
After speaking with @zramsay, I think we should use this time to actually simplify things and also not confuse other folks. So, let's rename:

`get-vendor-deps` => `deps`
`get-tools` => `tools`

Anything I'm missing @zramsay ?

**Comment by jackzampolin** [comment]
How about we merge these two PRs and then I'll make a PR to harmonize this across the repo?

**Comment by zramsay** [comment]
> Anything I'm missing @zramsay ?

nope :+1:

**Review by alexanderbez — APPROVED** [review]
LGTM given the aforementioned make targets will be renamed 👍

**Pull request merged**